packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -33,6 +43,18 @@ contract RepPriceOracle is IRepPriceOracle, Initializable {
         require(cash != address(0));
         uniswapFactory = IUniswapV2Factory(_augur.lookup("UniswapV2Factory"));
         require(uniswapFactory != IUniswapV2Factory(0));
+        ensRegistry = IENSRegistry(_augur.lookup("ENSRegistry"));
+        require(ensRegistry != IENSRegistry(0));
+    }
+
+    function upgradeUniswapFactory() public {
+        require(!uniswapUpgraded, "Already upgraded");
+        IENSResolver _resolver = IENSResolver(ensRegistry.resolver(UNISWAP_REGISTRY_ENS_NAME));
+        require(_resolver != IENSResolver(0), "Resolver for ENS name not found");
+        address _newUniswapFactory = _resolver.addr(UNISWAP_REGISTRY_ENS_NAME);
+        require(_newUniswapFactory != address(0), "Could not get address from resolver");
+        uniswapFactory = IUniswapV2Factory(_newUniswapFactory);
+        uniswapUpgraded = true;
     }
 
     // TODO: Consider when this should be called other than when the price is requested as part of new fee setting
```

Comment:
I _think_ that is the only time we need to call it if doing an exponential moving average?  For doing other types of moving averages it may need to be called on a regular interval (bot rewarded).

Reply:
Actually, you probably _should_ call this once every `period`.